### PR TITLE
[WIP] Integration of all EOS-S3 related changes/bugfixes

### PR DESCRIPTION
**Pull request by mkurc-ant**
This PR is intended to upstream all EOS-S3 related development which happened in the QuickLogic fork.

### CODE REVIEW

File: quicklogic/common/toolchain_wrappers/symbiflow_fasm2bels [hunk 1]
```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
```
Comment (mithro) [inline quicklogic/common/toolchain_wrappers/symbiflow_fasm2bels:1]: We shouldn't be writing long shell scripts like this.

File: quicklogic/pp3/tests/design_flow/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+add_subdirectory(design3)
+add_subdirectory(design6)
+add_subdirectory(design8)
+add_subdirectory(design9)
+add_subdirectory(e_sdio_host_controller)
+add_subdirectory(fifo_test)
+add_subdirectory(i2c_master_top)
+add_subdirectory(iir)
+add_subdirectory(inferred_ram_test)
+add_subdirectory(IR_Remote)
```
Comment (mithro) [inline quicklogic/pp3/tests/design_flow/CMakeLists.txt:25]:
```suggestion
add_subdirectory(ir_remote)
```

File: utils/yosys_fixup_cell_names.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
+#!/usr/bin/env python3
```
Comment (mithro) [inline utils/yosys_fixup_cell_names.py:1]: This script feels wrong. Why is it needed?

File: utils/vpr_fixup_post_synth.py [hunk 4]
```diff
@@ -76,6 +76,117 @@ def sub_func(match):
 # =============================================================================
 
 
+def merge_verilog_ports(code):
```
Comment (mithro) [inline utils/vpr_fixup_post_synth.py:79]: Why is this needed?

File: utils/report_timing.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
+import argparse
```
Comment (mithro) [inline utils/report_timing.py:1]: Missing documentation / license / etc...

**Review by mithro — CHANGES_REQUESTED** [review]
This is a massive pull request that is almost impossible to review.

**Comment by mkurc-ant** [comment]
@mithro Yes... I'm aware that this PR is far from mergable but issuing a PR was the only way to trigger Kokoro CI to test the changes. It exploded anyway.

**Pull request closed without merging**